src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -7117,7 +7117,7 @@ public override void VisitForEachIterationVariables(BoundForEachStatement node)
                         else if (node.Syntax is ForEachStatementSyntax { Type: { IsVar: true } })
                         {
                             // foreach (var variable in collection)
-                            _variableTypes[iterationVariable] = sourceState.ToTypeWithAnnotations();
+                            _variableTypes[iterationVariable] = sourceState.ToAnnotatedTypeWithAnnotations();
```

Comment:
>ToAnnotatedTypeWithAnnotations [](start = 76, length = 30)

We should consider not changing this one.  The `foreach` variable is readonly, so nobody's going to be assigning `null` to it.

Reply:
Discussed in more details in another thread in this PR. Tagged you there.